src/content/en/fundamentals/performance/http2/index.md

```diff
@@ -0,0 +1,568 @@
+intends to push to avoid creating own and duplicate requests for these
+resources. The simplest strategy to satisfy this requirement is to send all
+PUSH_PROMISE frames, which contain just the HTTP headers of the promised
+resource, ahead of the parent’s response (i.e., DATA frames).
+
+Once the client receives a PUSH_PROMISE frame it has the option to decline the
+stream (via a RST_STREAM frame) if it wants to (e.g., the resource is already in
+cache), which is an important improvement over HTTP/1.x. By contrast, the use of
+resource inlining, which is a popular "optimization" for HTTP/1.x, is equivalent
+to a "forced push": the client cannot opt-out, cancel it, or process the inlined
+resource individually.
+
+With HTTP/2 the client remains in full control of how server push is used. The
+client can limit the number of concurrently pushed streams; adjust the initial
+flow control window to control how much data is pushed when the stream is first
+opened; disable server push entirely. These preferences are communicated via the
+SETTINGS frames at the beginning of the HTTP/2 connection and may be updated at
+any time.
+
+Each pushed resource is a stream that, unlike an inlined resource, allows it to
+be individually multiplexed, prioritized, and processed by the client. The only
+security restriction, as enforced by the browser, is that pushed resources must
+obey the same-origin policy: the server must be authoritative for the provided
+content.
+
+## Header Compression
+
+Each HTTP transfer carries a set of headers that describe the transferred
+resource and its properties. In HTTP/1.x, this metadata is always sent as plain
+text and adds anywhere from 500–800 bytes of overhead per transfer, and
+sometimes kilobytes more if HTTP cookies are being used; see [Measuring and
+Controlling Protocol Overhead](https://hpbn.co/http1x/#measuring-and-
+controlling-protocol-overhead). To reduce this overhead and improve performance,
+HTTP/2 compresses request and response header metadata using the HPACK
+compression format that uses two simple but powerful techniques:
+
+1. It allows the transmitted header fields to be encoded via a static Huffman 
+   code, which reduces their individual transfer size.
+2. It requires that both the client and server maintain and update an indexed 
+   list of previously seen header fields (i.e., establishes a shared compression 
+   context), which is then used as a reference to efficiently encode previously 
+   transmitted values.
+
+Huffman coding allows the individual values to be compressed when transferred,
+and the indexed list of previously transferred values allows us to encode
+duplicate values by transferring index values that can be used to efficiently
+look up and reconstruct the full header keys and values.
+
+![](images/header_compression01.svg)
+
+As one further optimization, the HPACK compression context consists of a static
+and dynamic tables: the static table is defined in the specification and
```

Comment:
'a static and dynamic tables'

Not sure if this is a typo or intention. Are saying there is one static table and many dynamic tables or should the 'a' be removed.